docs/hugo/content/guide/asov1-asov2-migration/_index.md

```diff
@@ -0,0 +1,331 @@
+---
+title: ASOv1 to ASOv2 migration guide
+weight: 5
+layout: single
+cascade:
+  - type: docs
+  - render: always
+---
+ASOv1 is [deprecated](https://github.com/Azure/azure-service-operator/blob/main/README.md#aso-v1). 
+We strongly recommend migrating to ASOv2 to continue getting new features and support.
+
+## Assumptions
+
+- Migrating a single cluster.
+  - If you have ASOv1 installed on multiple clusters, apply these steps to each cluster.
+- ASOv1 is installed in its default namespace `azureoperator-system`.
+  - If it's installed into a different namespace, replace all occurrences of `azureoperator-system` in this document with 
+    the namespace it's installed into. 
+- ASOv2 is not yet installed in the cluster.
+- ASOv1 is managing a single subscription, so ASOv2 will also be managing that same single subscription via a single 
+  global credential.
+  - See [ASOv2 credential scope](../authentication/#credential-scope) for other credential scopes that allow for 
+    managing multiple subscriptions and/or using multiple identities.
+
+## Performing the migration
+
+### Check the version of cert-manager
+
+Ensure the version of cert-manager in your cluster supports `v1` resources.
+
+```
+kubectl api-resources --api-group cert-manager.io`
+```
+should show `v1` resources.
+
+> ℹ️ **Note:** We strongly recommend ensuring that you're running the latest version of cert-manager 
+> (1.14.4 at the time this article was written).
+
+> ℹ️ **Note:** We also recommend ensuring that ASOv1 is configured to use `cert-manager.io/v1` resources.
+> You can run `helm upgrade` and pass `--set certManagerResourcesAPIVersion=cert-manager.io/v1` to ensure ASOv1 is using
+> the v1 versions of the cert-manager resources.
```

Comment:
Does this do an upgrade, or just return an error if non v1 versions are in use?

Reply:
It depends a bit on what you mean by "upgrade". It doesn't change the versions of any resources in the cluster (container images, helm charts, etc), at least assuming you use the same version of the ASOv1 chart as you already have installed. What it does is delete the ASOv1 `v1alpha2` cert-manager CRs and replace them with `v1` cert-manager CRs.

So the net effect of this is:
- Delete v1alpha2 cert manager custom resources (not the CRDs, the CRs)
- Create v1 cert manager custom resources w/ same configuration as above.